pkg/virt-controller/watch/export/export.go

```diff
@@ -0,0 +1,866 @@
+						ClaimName: pvc.Name,
+					},
+				},
+			})
+		}
+		ctrl.addVolumeEnvironmentVariables(&podManifest.Spec.Containers[0], pvc, i, mountPoint)
+	}
+
+	// Add token and certs ENV variables
+	podManifest.Spec.Containers[0].Env = append(podManifest.Spec.Containers[0].Env, corev1.EnvVar{
+		Name:  "CERT_FILE",
+		Value: "/cert/tls.crt",
+	}, corev1.EnvVar{
+		Name:  "KEY_FILE",
+		Value: "/cert/tls.key",
+	}, corev1.EnvVar{
+		Name:  "TOKEN_FILE",
+		Value: "/token/token",
+	}, corev1.EnvVar{
+		Name:  "DEADLINE",
+		Value: currentTime().Add(deadline).Format(time.RFC3339),
+	})
+
+	secretName := fmt.Sprintf("secret-%s", rand.String(10))
+	podManifest.Spec.Volumes = append(podManifest.Spec.Volumes, corev1.Volume{
+		Name: certificates,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: secretName,
+			},
+		},
+	}, corev1.Volume{
+		Name: vmExport.Spec.TokenSecretRef,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: vmExport.Spec.TokenSecretRef,
+			},
+		},
+	})
+
+	podManifest.Spec.Containers[0].VolumeMounts = append(podManifest.Spec.Containers[0].VolumeMounts, corev1.VolumeMount{
+		Name:      certificates,
+		MountPath: "/cert",
+	}, corev1.VolumeMount{
+		Name:      vmExport.Spec.TokenSecretRef,
+		MountPath: "/token",
+	})
+	return podManifest
+}
+
+func (ctrl *VMExportController) addVolumeEnvironmentVariables(exportContainer *corev1.Container, pvc *corev1.PersistentVolumeClaim, index int, mountPoint string) {
+	exportContainer.Env = append(exportContainer.Env, corev1.EnvVar{
+		Name:  fmt.Sprintf("VOLUME%d_EXPORT_PATH", index),
+		Value: mountPoint,
+	})
+	if pvc.Spec.VolumeMode != nil && *pvc.Spec.VolumeMode == corev1.PersistentVolumeBlock {
+		exportContainer.Env = append(exportContainer.Env, corev1.EnvVar{
+			Name:  fmt.Sprintf("VOLUME%d_EXPORT_RAW_URI", index),
+			Value: rawURI(pvc),
+		}, corev1.EnvVar{
+			Name:  fmt.Sprintf("VOLUME%d_EXPORT_RAW_GZIP_URI", index),
+			Value: rawGzipURI(pvc),
+		})
+	} else {
+		if ctrl.isKubevirtContentType(pvc) {
+			exportContainer.Env = append(exportContainer.Env, corev1.EnvVar{
+				Name:  fmt.Sprintf("VOLUME%d_EXPORT_RAW_URI", index),
+				Value: rawURI(pvc),
+			}, corev1.EnvVar{
+				Name:  fmt.Sprintf("VOLUME%d_EXPORT_RAW_GZIP_URI", index),
+				Value: rawGzipURI(pvc),
+			})
+		} else {
+			exportContainer.Env = append(exportContainer.Env, corev1.EnvVar{
+				Name:  fmt.Sprintf("VOLUME%d_EXPORT_ARCHIVE_URI", index),
+				Value: archiveURI(pvc),
+			}, corev1.EnvVar{
+				Name:  fmt.Sprintf("VOLUME%d_EXPORT_DIR_URI", index),
+				Value: dirURI(pvc),
+			})
+		}
+	}
+}
+
+func (ctrl *VMExportController) isKubevirtContentType(pvc *corev1.PersistentVolumeClaim) bool {
+	ann := pvc.GetAnnotations()
+	if ann == nil {
+		return true
+	}
+	value, ok := ann[annContentType]
+	return !ok || value == string(cdiv1.DataVolumeKubeVirt) || value == ""
```

Comment:
I think this should only return true if the annotation exists and equals `kubevirt`.  As-is we will serve up memdump pvs as kubevirt content type

Reply:
I am following the logic CDI uses here, if not specified then its kubevirt, we have many PVCs without any content type annotation, which would be served as archive here.

Reply:
I did some checking and DVs will actually add the content type annotation to the PVC, so we can assume it is there for purposes of export.

Reply:
Found an exception to this, importing from registry does not seem to add the kubevirt content type annotation to the pvc. Not sure why but this means there are many pvcs out there without this annotation.

Reply:
CDI is writing data.  Export is reading data.  I think it makes sense to use the more generic access method (Dir) unless we know for sure it is kubevirt